### ISSUE TITLE
Checker for `TypeVar` name and first argument

### ISSUE THREAD

**Issue by DanielNoord** [PyCQA/pylint#5224]
### Current problem

Maybe more an idea for a separate checker. `TypeVar`, `NewType`, and `ParamSpec` have special name requirements: The first argument should match the variable name. Something like this is discouraged
```py
T = TypeVar("T_Something")
```

Would be nice if we could also detect it, too.

_Originally posted by @cdce8p in https://github.com/PyCQA/pylint/pull/5221#discussion_r737677291_

### Desired solution

Basically:

```python
T = TypeVar("T", ...) # good
K = TypeVar("T", ...) # bad
K_contra = TypeVar("T", ...) # bad
```

**Comment by jpy-git** [comment]
_From #6011_
```python
T = TypeVar('A')
```
Suggested name: `typevar-name-mismatch`
Suggested message: `TypeVar must be assigned to a variable named "A"`
